spec/calendar.html

```diff
@@ -483,16 +483,50 @@ <h1>ISODaysInMonth ( _year_, _month_ )</h1>
+        <dd>It implements the logic for the `Temporal.Calendar.prototype.weekOfYear` method, which is used for the ISO 8601 calendar.</dd>
+      </dl>
       <emu-alg>
-        1. Assert: _year_ is an integer.
-        1. Assert: _month_ is an integer.
-        1. Assert: _day_ is an integer.
-        1. Let _date_ be the date given by _year_, _month_, and _day_.
-        1. Return _date_'s week number according to ISO-8601 as an integer.
+        1. Assert: IsValidISODate(_year_, _month_, _day_) is *true*.
+        1. Let _wednesday_ be 3.
+        1. Let _thursday_ be 4.
+        1. Let _friday_ be 5.
+        1. Let _saturday_ be 6.
+        1. Let _daysInWeek_ be 7.
+        1. Let _maxWeekNumber_ be 53.
+        1. Let _dayOfYear_ be ToISODayOfYear(_year_, _month_, _day_).
+        1. Let _dayOfWeek_ be ToISODayOfWeek(_year_, _month_, _day_).
+        1. Let _week_ be floor((_dayOfYear_ - (_daysInWeek_ - _dayOfWeek_ + _wednesday_ )) / _daysInWeek_).
+        1. If _week_ &lt; 1, then
+          1. NOTE: This is the last week of the previous year.
+          1. Let _dayOfJan1st_ be ToISODayOfWeek(_year_, 1, 1).
+          1. If _dayOfJan1st_ is _friday_, then
+            1. Return _maxWeekNumber_.
+          1. If _dayOfJan1st_ is _saturday_, and InLeapYear(TimeFromYear(𝔽(_year_ - 1))) is *1*<sub>𝔽</sub>, then
+            1. Return _maxWeekNumber_.
+          1. Return _maxWeekNumber_ - 1.
+        1. If _week_ is _maxWeekNumber_, then
+          1. Let _daysInYear_ be DaysInYear(𝔽(_year_)).
+          1. Let _daysLaterInYear_ be _daysInYear_ - _dayOfYear_.
+          1. Let _daysAfterThursday_ be _thursday_ - _dayOfWeek_.
+          1. If _daysLaterInYear_ &lt; _daysAfterThursday_, then
+            1. Return 1.
+        1. Return _week_.
       </emu-alg>
       <emu-note>Beware that dates at the beginning of a year may be part of a week from the preceding year, and dates at the end of a year may be part of a week at the beginning of the next year, as the first week of any year is defined as the week that contains the first Thursday of the year.</emu-note>
+      <emu-note>This rule is based on the principle that a week belongs to the calendar year to which the majority of its calendar days belong.</emu-note>
+      <emu-note>In the week calendar, calendar days of the first and last calendar week of a calendar year may belong to the previous and the next calendar year respectively.</emu-note>
+      <emu-note>ISO week 1 of every calendar year includes its January 4 and starts on the first Monday on or before that, the ISO week calendar year of any date is its calendar year adjusted by -1 if it precedes that Monday or by 1 if it falls on or after that Monday of week 1 for the next calendar year, and the ISO week number of any date is 1 plus the number of complete weeks from the start of its week calendar week 1.</emu-note>
+      <emu-note>See ISO 8601-1:2019 section 3.1.1.23 week calendar for more details.</emu-note>
```

Comment:
```suggestion
      <emu-note>In the ISO 8601 week calendar, calendar week number 1 of a calendar year is the week including the first Thursday of that year (based on the principle that a week belongs to the same calendar year as the majority of its calendar days), which always includes January 4 and starts on the Monday on or immediately before then. Because of this, some calendar days of the first calendar week of a calendar year may be part of the _preceding_ [proleptic Gregorian] date calendar year, and some calendar days of the last calendar week of a calendar year may be part of the _following_ [proleptic Gregorian] date calendar year. See ISO 8601 for details.</emu-note>
      <emu-note>For example, week calendar year 2020 includes both 31 December 2019 (a Tuesday belonging to its calendar week 1) and 1 January 2021 (a Friday belonging to its calendar week 53).</emu-note>
```